ArduPlane/navigation.cpp

```diff
@@ -208,6 +208,28 @@ void Plane::calc_airspeed_errors()
                 target_airspeed_cm = aparm.airspeed_cruise_cm;
             }
         }
+
+#if HAL_SOARING_ENABLED
+    } else if (g2.soaring_controller.is_active() && g2.soaring_controller.get_throttle_suppressed()) {
+        if (control_mode == &mode_thermal) {
+            float arspd = g2.soaring_controller.get_thermalling_target_airspeed();
+
+            if (arspd > 0) {
```

Comment:
On vacation on the road and haven't had a chance to look at any of the details but the question comes to mind: does all of this depend on having an AirSpeed sensor and what if you don't?

Reply:
@magicrub the target airspeed gets constrained to those limits at the end of the function
@Hwurzburg although right now soaring officially requires an airspeed sensor it will happily use the AHRS airspeed estimate . . . it's a different matter whether this will work well - there's a separate effort ongoing to improve this